virtualizarr/manifests/array_api.py

```diff
@@ -100,12 +102,12 @@ def concatenate(
         lengths=concatenated_lengths,
     )
 
-    # chunk shape has not changed, there are just now more chunks along the concatenation axis
-    new_zarray = first_arr.zarray.replace(
-        shape=tuple(new_shape),
-    )
+    metadata_copy = first_arr.metadata.to_dict().copy()
```

Comment:
This seems overly verbose, and something we will want in multiple places. Let's make a `replace` constructor function, before adding a method upstream.

Reply:
https://github.com/zarr-developers/VirtualiZarr/pull/429/commits/aad511f86ff1e89dd4cf6b37252bf6adf3d3d289